src/System.Private.CoreLib/src/System/StubHelpers.cs

```diff
@@ -1715,14 +1720,28 @@ internal static void CheckStringLength(int length)
 
         internal static void CheckStringLength(uint length)
         {
-            if (length > 0x7ffffff0)
+            if (length > MaxStringLength)
             {
-                throw new MarshalDirectiveException(SR.Marshaler_StringTooLong);
+                ThrowStringTooLong();
             }
         }
 
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        internal static extern unsafe int strlen(sbyte* ptr);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static unsafe int strlen(byte* ptr)
```

Comment:
I am wondering whether it would be better for this to live in String.cs next to `wcslen`; or in some other place not related to index (maybe SpanHelpers?).

Reply:
Moved to `string`